main.nf

```diff
@@ -12,251 +12,27 @@
-      --maltextract_percentidentity [num]       Minimum percent identity alignments are required to have to be reported. Recommended to set same as MALT parameter. Default: ${params.maltextract_percentidentity}
-      --maltextract_topalignment [int]          Turn on using top alignments per read after filtering.
-
-    Other options:     
-      -name [str]                     Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
-      --max_memory [str]              Memory limit for each step of pipeline. Should be in form e.g. --max_memory '8.GB'. Default: '${params.max_memory}'
-      --max_time [str]                Time limit for each step of the pipeline. Should be in form e.g. --max_time '2.h'. Default: '${params.max_time}'
-      --max_cpus [str]                Maximum number of CPUs to use for each step of the pipeline. Should be in form e.g. Default: '${params.max_cpus}'
-      --publish_dir_mode [str]        Mode for publishing results in the output directory. Available: symlink, rellink, link, copy, copyNoFollow, move. Default: '${params.publish_dir_mode}'
-      --email [email]                 Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits
-      --email_on_fail [email]         Same as --email, except only send mail if the workflow is not successful
-      --plaintext_email [email]       Receive plain text emails rather than HTML
-      --max_multiqc_email_size [str]  Threshold size for MultiQC report to be attached in notification email. If file generated by pipeline exceeds the threshold, it will not be attached (Default: 25MB)
-
-    AWSBatch options:
-      --awsqueue [str]   The AWSBatch JobQueue that needs to be set when running on AWSBatch
-      --awsregion [str]  The AWS Region for your AWS Batch job to run on
-      --awscli [str]     Path to the AWS CLI tool
-      
-    For a full list and more information of available parameters, consider the documentation (https://github.com/nf-core/eager/).
-    """.stripIndent()
-}
-
-///////////////////////////////////////////////////////////////////////////////
-/* --                SET UP CONFIGURATION VARIABLES                       -- */
-///////////////////////////////////////////////////////////////////////////////
-
 // Show help message
 params.help = false
-if (params.help){
-    helpMessage()
+def json_schema = "$projectDir/nextflow_schema.json"
+if (params.help) {
+    def command = "nextflow run nf-core/eager -profile <docker/singularity/conda> --reads'*_R{1,2}.fastq.gz' --fasta '<your_reference>.fasta'"
+    log.info NfcoreSchema.params_help(workflow, params, json_schema, command)
     exit 0
 }
 
+////////////////////////////////////////////////////
+/* --         VALIDATE PARAMETERS              -- */
+////////////////////////////////////////////////////
+
+def unexpectedParams = []
+if (params.validate_params) {
```

Comment:
Is this param something one has to manually add at CLI execution?

EDIT: Ah ok, set to true by default :+1: